wasm/datatypes/configuration.py

```diff
@@ -0,0 +1,61 @@
+from typing import (
+    NamedTuple,
+)
+
+from .instructions import (
+    InstructionSequence,
+)
+from .stack import (
+    Frame,
+    FrameStack,
+    Label,
+    LabelStack,
+    ValueStack,
+)
+from .store import (
+    Store,
+)
+
+
+class Configuration(NamedTuple):
```

Comment:
Some concerns about this class:

* Its definition and function seem to diverge a bit from the spec.  Namely, configurations in the spec are just a tuple of references to some current *store*, *frame*, and *instruction sequence*.  Then all the instructions essentially act on a mutable configuration.
* I think the separation of the execution stack into three homogeneously typed stacks might be problematic.  This might account for some of the difficulty in tracking which labels belong to which frame.  In terms of the semantics defined by the spec, I believe it's correct to say that all labels that appear immediately above a frame in the execution stack belong to that frame.  It's then easy to pop off a frame's labels; you just pop until you hit a frame.  Also, the semantics of the reduction rules are, I think, fairly well-coupled with the notion of a single, heterogeneous stack.

The above concerns are tied in with my intuition that it will be very useful if the implementation can correspond as much as possible with the spec.  That should help new developers a lot.